src/ui/connect/connector.ts

```diff
@@ -1,107 +1,127 @@
+import {isChromium, isFirefox} from 'utils/platform';
 import type {ExtensionData, ExtensionActions, FilterConfig, TabInfo, Message, UserSettings} from '../../definitions';
 
 export default class Connector implements ExtensionActions {
-    private port: chrome.runtime.Port;
-    private counter: number;
+    private changeSubscribers: Set<(data: ExtensionData) => void>;
 
     constructor() {
-        this.counter = 0;
-        this.port = chrome.runtime.connect({name: 'ui'});
+        this.changeSubscribers = new Set();
     }
 
-    private getRequestId() {
-        return ++this.counter;
+    private async sendRequest<T>(request: Message) {
+        return new Promise<T>((resolve, reject) => {
+            chrome.runtime.sendMessage({name: 'ui', ...request}, ({data, error}) => {
+                if (error)
+                    reject(error);
+                else
+                    resolve(data);
+            });
+        });
     }
 
-    private async sendRequest<T>(request: Message, executor: (response: Message, resolve: (data?: T) => void, reject: (error: Error) => void) => void) {
-        const id = this.getRequestId();
+    private async poorMansSendRequest<T>(name: string) {
```

Comment:
`poorMansSendRequest` 😄 what should the naming represents?

Reply:
I renamed it to `firefoxSendRequestWithResponse` to match actual use of the function.